internal/server/data/data.go

```diff
@@ -155,7 +155,18 @@ func save[T models.Modelable](db *gorm.DB, model *T) error {
 }
 
 func add[T models.Modelable](db *gorm.DB, model *T) error {
-	err := db.Create(model).Error
+	var err error
```

Comment:
I didn't add this rollback logic to `save` because I couldn't think of a scenario we would expect an error when calling that function. If you can think of something let me know and I'll add it to `save` to.